webapp/Controllers/OrganizationsController.cs

```diff
@@ -0,0 +1,215 @@
+            _logger.LogInformation($"New organization added with ID: {org.Id}");
+            _logger.LogInformation($"New EmployeeOrganization added with ID: {user.Id}");
+
+            return RedirectToAction(nameof(Created));
+            
+        }
+        catch (Exception e)
+        {
+            _logger.LogWarning($"Error occured while creating organization:\n{e.Message}");
+            return StatusCode(500, new { errorMessage = e.Message });
+        }        
+    }
+
+    [HttpGet]
+    public async Task<IActionResult> Update(Guid id)
+    {
+        var existingOrg = await _ctx.Organizations.FirstOrDefaultAsync(o => o.Id == id);
+        if(existingOrg != default)
+        {
+            return View(existingOrg.ToOrgModel());
+        }
+        else
+        {
+            return RedirectToAction(nameof(Created));
+        }
+    }
+
+    [HttpPost("{id}/update")]
+    public async Task<IActionResult> UpdateModel(Guid id, UpdateOrganizationViewModel model)
+    {
+        if(!ModelState.IsValid)
+        {
+            _logger.LogInformation($"Model validation failed for {JsonSerializer.Serialize(model)}");
+            return RedirectToAction("Created");
+        }
+       
+        var existingOrg = await _ctx.Organizations.FirstOrDefaultAsync(o => o.Id == id);
+        if(existingOrg == null)
+        {
+            return NotFound();
+        }        
+        
+        try
+        {
+            existingOrg.Address = model.Address;
+            existingOrg.Phone = model.Phone;
+            existingOrg.Email = model.Email; 
+            existingOrg.Name = model.Name; 
+                    
+            await _ctx.SaveChangesAsync();  
+            
+            _logger.LogInformation($"Organization updated with ID: {existingOrg.Id}");  
+            
+            return RedirectToAction(nameof(Created));
+            
+        }
+        catch (Exception e)
+        {
+            _logger.LogWarning($"Error occured while updating organization:\n{e.Message}");
+            return StatusCode(500, new { errorMessage = e.Message });
+        }
+    }
+
+    [HttpPost("{id}/delete")]
+    public async Task <IActionResult> Delete(Guid id)
+    {
+        if (!await _ctx.Organizations.AnyAsync(p => p.Id == id))
+        {
+            return NotFound();
+        }
+
+        var org = await _ctx.Organizations.FirstOrDefaultAsync(y => y.Id == id);
+
+        try
+        {
+           _ctx.Organizations.Remove(org);
+
+            await _ctx.SaveChangesAsync();
+
+            _logger.LogInformation($"Organization deleted with ID: {org.Id}");
+
+            return RedirectToAction("list");            
+        }
+        catch (Exception e)
+        {
+            _logger.LogWarning($"Error occured while updating organization:\n{e.Message}");
+            return StatusCode(500, new { errorMessage = e.Message });
+        }
+    }
+
+    [HttpGet]
+    public IActionResult New() => View(new CreateOrganizationViewModel());
```

Comment:
This action should go upto right before `Create` action.

Reply:
bajarildi